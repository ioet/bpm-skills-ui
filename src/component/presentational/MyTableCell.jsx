import React from 'react';
import PropTypes from 'prop-types';
import withStyles from '@material-ui/core/styles/withStyles';
import { TableCellStyles } from '../../styles';

const MyTableCell = (props) => {
  const {
    classes, onMouseOver, onMouseOut, children,
  } = props;

  return (
    <div
      className={classes.cell}
      onFocus={(e) => {
        e.preventDefault();
        onMouseOver();
      }}
      onMouseOver={(e) => {
        e.preventDefault();
        onMouseOver();
      }}
      onMouseOut={onMouseOut}
      onBlur={onMouseOut}
    >
      {children}
    </div>
  );
};

MyTableCell.propTypes = {
  classes: PropTypes.object.isRequired,
  onMouseOver: PropTypes.func.isRequired,
  onMouseOut: PropTypes.func.isRequired,
  children: PropTypes.element.isRequired,
};

export default withStyles(TableCellStyles)(MyTableCell);
