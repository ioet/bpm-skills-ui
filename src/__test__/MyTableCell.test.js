import React from 'react';
import MyTableCell from '../component/presentational/MyTableCell';

describe('<MyTableCell />', () => {
  test('Component Renders correctly', () => {
    const appWrapper = shallow(<MyTableCell />);

    expect(toJson(appWrapper))
      .toMatchSnapshot();
  });
});
