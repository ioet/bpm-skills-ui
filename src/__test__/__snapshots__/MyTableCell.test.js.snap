// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyTableCell /> Component Renders correctly 1`] = `
<MyTableCell
  classes={
    Object {
      "cell": "MyTableCell-cell-1",
    }
  }
/>
`;
